hbase-server/src/main/java/org/apache/hadoop/hbase/master/assignment/MergeTableRegionsProcedure.java

```diff
@@ -540,7 +541,10 @@ private void preMergeRegions(final MasterProcedureEnv env) throws IOException {
     }
     // TODO: Clean up split and merge. Currently all over the place.
     try {
-      env.getMasterServices().getMasterQuotaManager().onRegionMerged(this.mergedRegion);
+      MasterQuotaManager masterQuotaManager = env.getMasterServices().getMasterQuotaManager();
```

Comment:
Hi @mnpoonia in which case are you seeing this NPE? Master is starting up and quotas has not yet initialized?

Reply:
@NihalJain You are right. A Master was in transition when we got this exception. It was trying to become active. Just after this NPE i can see a log line 

```
2024-10-14 20:02:11,265 ERROR [PEWorker-31] procedure2.ProcedureExecutor - CODE-BUG: Uncaught runtime exception: pid=58745238, state=RUNNABLE:SPLIT_TABLE_REGION_PRE_OPERATION, locked=true; SplitTableRegionProcedure table=Table1, parent=36dcdeb9044bd0ece2eb046cf7fd7c2f, daughterA=cfba8d001ef9d3640060a4e845d3d013, daughterB=0becd69b7ccdc7e12dcc96cc637dc300
java.lang.NullPointerException

2024-10-14 20:02:11,269 INFO  [aster/hmaster-1:61000:becomeActiveMaster] quotas.MasterQuotaManager - Quota support disabled
```